File: ShoppingList/src/main/java/org/openintents/shopping/sync/NoSyncSupport.java
package org.openintents.shopping.sync;

import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

import org.openintents.shopping.SyncSupport;

public class NoSyncSupport implements SyncSupport {
    @Override
    public boolean isAvailable() {
        return false;
    }

    @Override
    public void pushListItem(long listId, Cursor cursor) {

    }

    @Override
    public void pushList(Cursor cursor) {

    }

    @Override
    public boolean isSyncEnabled() {
        return false;
    }

    @Override
    public void setSyncEnabled(boolean enableSync) {

    }

    @Override
    public void updateListItem(long listId, Uri itemUri, ContentValues values) {

    }

}

File: ShoppingList/src/main/java/org/openintents/shopping/sync/NoSyncSupport.kt
package org.openintents.shopping.sync

import android.content.ContentValues
import android.database.Cursor
import android.net.Uri
import org.openintents.shopping.SyncSupport

class NoSyncSupport : SyncSupport {

    override fun isAvailable(): Boolean = false

    override fun pushListItem(listId: Long, cursor: Cursor?) {
    }

    override fun pushList(cursor: Cursor?) {
    }

    override fun isSyncEnabled(): Boolean = false

    override fun setSyncEnabled(enableSync: Boolean) {
    }

    override fun updateListItem(listId: Long, itemUri: Uri?, values: ContentValues?) {
    }
}
